lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -374,13 +374,13 @@ def error_code_from(response)
 
       # This implementation follows the Cybersource guide on how create the request signature, see:
       # https://developer.cybersource.com/docs/cybs/en-us/payments/developer/all/rest/payments/GenerateHeader/httpSignatureAuthentication.html
-      def get_http_signature(resource, digest, http_method = 'post', gmtdatetime = Time.now.httpdate)
+      def get_http_signature(resource, digest, options, http_method = 'post', gmtdatetime = Time.now.httpdate)
         string_to_sign = {
           host: host,
           date: gmtdatetime,
           "(request-target)": "#{http_method} /pts/v2/#{resource}",
           digest: digest,
-          "v-c-merchant-id": @options[:merchant_id]
+          "v-c-merchant-id": options[:merchant_id] || @options[:merchant_id]
```

Comment:
After reviewing the [REST docs](https://developer.cybersource.com/docs/cybs/en-us/security-keys/user/all/ada/security-keys/keys-meta-intro/keys-meta-api-intro.html) I made a mistake in the requirements. I believe we should only override it for the HTTP header, not the Signature.